api/error.go

```diff
@@ -42,6 +42,24 @@ func (e Error) LogData() map[string]interface{} {
 	return e.logData
 }
 
+// notFoundErr satisfies the `errUnavailable` interface
```

Comment:
I think it would be better to keep to how the error types are implemented in the mongo package, i.e. add the NotFound tags and functions to the main error type. Not only will it be consistent but this way you can return logData at the same time etc 
```suggestion
// Error is the packages error type
type Error struct {
	err     error
	message string
	logData map[string]interface{}
	notFound bool
	badRequest bool
}

func (e err) NotFound() bool{
  return e.notFound
}
```
etc

Reply:
Makes sense to be consistent, have updated 👍. My qualm with a shared type and multiple flags is that it's usually the result of trying to represent a union — it's unlikely you'd want an error that is both `notFound` and `badRequest`, but unfortunately the type system permits it if you go this route.

Reply:
Yeah it does leave that room for programmer error.. At the moment the way it resolves if does happen that there's a hierarchy between the codes. It could be that we could implement an enum for the underlying values rather than bools, so long as we're still using the individual interfaces to determine the result.